c/datasetjson.c

```diff
@@ -1756,7 +1759,27 @@ void respondWithDatasetMetadata(HttpResponse *response) {
 
   int fieldCount = defaultCSIFieldCount;
   char **csiFields = defaultCSIFields;
-
+  int asteriskPos = indexOf(dsn,dsnLen,'*',0);
+  int dblAsteriskPos = indexOfString(dsn,dsnLen,"**",0);
+  int periodPos = lastIndexOf(dsn, dsnLen, '.');
+  int addQualifiers = !strcmp(addQualifiersArg, "true");
+  
+  if(addQualifiers){
+	  int newDsnLen = dsnLen + 4; //+2 for asterisk, +1 for period, +1 for null terminator
+	  char newDsn[newDsnLen];
+	  strncpy(newDsn, dsn, dsnLen + 1);
```

Comment:
Doing a snprintf might be more elegant than doing a strncpy and then concatenating.